cmd/dra-example-controller/driver.go

```diff
@@ -81,22 +85,45 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 	case gpucrd.GpuClaimParametersKind:
 		gc, err := d.clientset.GpuV1alpha1().GpuClaimParameters(claim.Namespace).Get(ctx, claim.Spec.ParametersRef.Name, metav1.GetOptions{})
 		if err != nil {
-			return nil, fmt.Errorf("error getting GpuClaimParameters called '%v' in namespace '%v': %v", claim.Spec.ParametersRef.Name, claim.Namespace, err)
+			return nil, fmt.Errorf(
+				"error getting GpuClaimParameters called '%v' in namespace '%v': %v",
+				claim.Spec.ParametersRef.Name, claim.Namespace, err)
 		}
 		err = d.gpu.ValidateClaimParameters(&gc.Spec)
 		if err != nil {
-			return nil, fmt.Errorf("error validating GpuClaimParameters called '%v' in namespace '%v': %v", claim.Spec.ParametersRef.Name, claim.Namespace, err)
+			return nil, fmt.Errorf(
+				"error validating GpuClaimParameters called '%v' in namespace '%v': %v",
+				claim.Spec.ParametersRef.Name, claim.Namespace, err)
 		}
 		return &gc.Spec, nil
 	}
 	return nil, fmt.Errorf("unknown ResourceClaim.ParametersRef.Kind: %v", claim.Spec.ParametersRef.Kind)
 }
 
-func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
+func (d driver) Allocate(
+	ctx context.Context,
+	claim *resourcev1.ResourceClaim,
+	claimParameters interface{},
+	class *resourcev1.ResourceClass,
+	classParameters interface{},
+	selectedNode string) (*resourcev1.AllocationResult, error) {
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		return d.allocateImmediateClaim(claim, claimParameters, class, classParameters)
 	}
 
+	if err := d.allocateDelayedClaim(claim, claimParameters, class, classParameters, selectedNode); err != nil {
+		return nil, err
+	}
+
+	return buildAllocationResult(selectedNode, true), nil
+}
+
+func (d driver) allocateDelayedClaim(
+	claim *resourcev1.ResourceClaim,
+	claimParameters interface{},
+	class *resourcev1.ResourceClass,
+	classParameters interface{},
+	selectedNode string) error {
```

Comment:
Let's undo all of these, and then I'll do another pass on the PR

Reply:
Done.